var/spack/repos/builtin/packages/py-pyqt5/package.py

```diff
@@ -69,15 +70,15 @@ def make_qsci(self):
                 pydir = join_path(site_packages_dir, 'PyQt5')
                 python = self.spec['python'].command
                 python('configure.py', '--pyqt=PyQt5',
-                       '--sip=' + self.prefix.bin.sip,
+                       '--sip=' + self.spec['py-sip'].prefix.bin.sip,
                        '--qsci-incdir=' +
                        self.spec['qscintilla'].prefix.include,
                        '--qsci-libdir=' + self.spec['qscintilla'].prefix.lib,
                        '--qsci-sipdir=' + self.prefix.share.sip.PyQt5,
                        '--apidir=' + self.prefix.share.qsci,
                        '--destdir=' + pydir,
                        '--pyqt-sipdir=' + self.prefix.share.sip.PyQt5,
-                       '--sip-incdir=' + python_include_dir,
+                       '--sip-incdir=' + self.spec['py-sip'].prefix.include,
```

Comment:
@Sinan81 can you review these changes? I don't even remember how SIP works anymore.

Reply:
Sure. I will take a look at it by tomorrow at the latest.

Reply:
This is related to #14121

Reply:
@glennpj `self.site_packages_dir` thing yields "no such attribute" errors. I think you might want to revert that modification or else I am missing something.

Reply:
As expected, the suggested modification is causing build errors for qsci variant:
```
spack install py-pyqt5+qsci
...
  >> 4611    sipAPIQsci.h:31:10: fatal error: sip.h: No such f
             ile or directory
     4612     #include <sip.h>
     4613              ^~~~~~~
     4614    In file included from sipQsciQsciCommand.cpp:24:0
             :
  >> 4615    sipAPIQsci.h:31:10: fatal error: sip.h: No such f
             ile or directory
     4616     #include <sip.h>
     4617              ^~~~~~~
     4618    compilation terminated.
```

Reply:
> @glennpj `self.site_packages_dir` thing yields "no such attribute" errors. I think you might want to revert that modification or else I am missing something.

Well, something is not right. If using `site_packages_dir` then the following happens when regenerating the module:
```console
$ spack module tcl refresh py-pyqt5
==> You are about to regenerate tcl module files for:

-- linux-centos7-broadwell / gcc@9.1.0 --------------------------
kqaagrv py-pyqt5@5.13.0

==> Do you want to proceed? [y/n] y
==> Regenerating tcl module files
==> Warning: Could not write module file [/opt/packages/gpjohnsn/spack/share/spack/modules/linux-centos7-broadwell/py-pyqt5-5.13.0-gcc-9.1.0-kqaagrv]
==> Warning:    --> name 'site_packages_dir' is not defined <--
```
That may be a separate issue from this package, although I have not seen that behavior with anything else.

Reply:
> As expected, the suggested modification is causing build errors for qsci variant:

Yes, I seem to have gotten something confused somewhere.

Reply:
@adamjstewart might be able to help with the 'site_packages_dir' issue above. I've never seen that before either.

Reply:
I just pushed a commit to undo the changes to `py-pyqt5`. The only issue remaining is the `site_packages_dir` issue. I can no longer replicate what I was chasing down before so I am guessing that I did something silly along the way.